force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -147,19 +162,59 @@ export default class ServiceDeliveryRow extends LightningElement {
             });
     }
 
+    handleGetServicesForProgramEngagement(programEngagementId) {
+        this.isError = false;
+
+        getServicesByProgramEngagementId({ programEngagementId: programEngagementId })
+            .then(result => {
+                if (result) {
+                    this._filteredServices = result[SERVICES].slice(0);
+                } else {
+                    this.isError = true;
+                }
+
+                this.enableServiceInputWithOptions();
+            })
+            .catch(err => {
+                this.rowError = handleError(err, false);
+            });
+    }
+
     handleInputChange(event) {
+        console.log(event.target.fieldName);
         if (event.target.fieldName === this.fields.contact.fieldApiName) {
             if (event.detail.value && event.detail.value.length) {
                 this.selectedContact = event.detail.value[0];
                 this.handleGetServicesEngagements(event.detail.value[0]);
             } else {
                 this.handleResetContact();
             }
+        } else if (
+            event.target.fieldName === this.fields.programEngagement.fieldApiName
+        ) {
+            this.isServiceFiltered = true;
+            this.handleGetServicesForProgramEngagement(event.detail.value[0]);
+        } else if (event.target.fieldName === this.fields.service.fieldApiName) {
+            this.handleEnableFieldOnInputChange(event.target.fieldName);
+            this.autoSaveAfterDebounce();
         } else {
             this.autoSaveAfterDebounce();
         }
     }
 
+    handleEnableFieldOnInputChange(fieldApiName) {
```

Comment:
Is there a better spot to clone the fieldset object, so it's not done repeatedly on input enable?  Feels like we should only have to do this once.

Reply:
I tried it doing it earlier and not having to do it again but it did not work. I can try it again. I think the problem is that we run into this issue only when we are changing a value of a lookup and trying to update the fieldset. I will give it another try and see if it works if we try to clone it earlier and not have to do it everytime.